tools/concat_multi_datasets/catWrapper.xml

```diff
@@ -1,43 +1,224 @@
+            <param name="inputs" value="1.bed,2.bed"/>
+            <param name="dataset_names" value="No" />
+            <param name="headers" value="0" />
             <output name="out_file1" file="cat_wrapper_out1.bed"/>
         </test>
-        <test>
-            <param name="headers" value="Yes" />
-            <param name="input" value="1.bed,2.bed"/>
+        <test> <!-- Test 2 single files concatenation with dataset names activated -->
+            <param name="input_type" value="singles" />
+            <param name="inputs" value="1.bed,2.bed"/>
+            <param name="dataset_names" value="Yes" />
+            <param name="headers" value="0" />
             <output name="out_file1" file="cat_wrapper_out2.bed"/>
         </test>
-
+        <test> <!-- Test 2 single files concatenation skipping 1 line -->
+            <param name="input_type" value="singles" />
+            <param name="inputs" value="1.bed,2.bed"/>
+            <param name="dataset_names" value="No" />
+            <param name="headers" value="1" />
+            <output name="out_file1" file="cat_wrapper_out3.bed"/>
+        </test>
+        <test> <!-- Test gz handling with no options -->
+            <param name="input_type" value="singles" />
+            <param name="inputs" value="1_f.fastq.gz,1_r.fastq.gz"/>
+            <param name="dataset_names" value="No" />
+            <param name="headers" value="0" />
+            <output name="out_file1" file="1.fastq.gz" decompress="True"/>
+        </test>
+        <test> <!-- Test gz handling with options -->
+            <param name="input_type" value="singles" />
+            <param name="inputs" value="1_f.fastq.gz,1_r.fastq.gz"/>
+            <param name="dataset_names" value="Yes" />
+            <param name="headers" value="4" />
+            <output name="out_file1" file="1_options.fastq.gz" decompress="True"/>
+        </test>
+        <!-- Test paired options -->
+        <test> <!-- Test paired gziped collection concatenation by_pair with no other option -->
+            <param name="input_type" value="paired_collection" />
+            <param name="paired_cat_type" value="by_pair"/>
+            <param name="inputs">
+                <collection type="list:paired">
+                    <element name="1">
+                        <collection type="paired">
+                            <element name="forward" value="1_f.fastq.gz"/>
+                            <element name="reverse" value="1_r.fastq.gz"/>
+                        </collection>
+                    </element>
+                </collection>
+            </param>
+            <param name="dataset_names" value="No" />
+            <param name="headers" value="0" />
+            <output_collection name="list_output" type="list" >
+                <element name="1" file="1.fastq.gz" decompress="True"/>
+            </output_collection>
+        </test>
+        <!-- Test nested -->
     </tests>
     <help>
```

Comment:
The help section is far too short. There must be already 2 or 3 concatenation tools for galaxy. What this one is doing is not explained, except for the simplest case.